### PR TITLE
Updated use-ncnn-with-pytorch-or-onnx document

### PR DESCRIPTION
**Pull request by whyb**
### Updated use-ncnn-with-pytorch-or-onnx document
* Recommended to use the onnxslim tool
* Recommended to use the pnnx tool

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5557) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5557) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: nihui<br/>:x: whyb<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=5557) it.</sub>

**Comment by whyb** [comment]
@nihui updated

File: docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md [hunk 1]
```diff
@@ -22,9 +128,10 @@ x = torch.rand(1, 3, 224, 224)
 # Export the model
 torch_out = torch.onnx._export(model, x, "resnet18.onnx", export_params=True)
 ```
+</details>
 
 ## simplify onnx model
```
Comment (nihui) [inline docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md:133]:
```suggestion
## simplify onnx model (deprecated)
```

File: docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md [hunk 2]
```diff
@@ -2,8 +2,114 @@ Here is a practical guide for converting pytorch model to ncnn
...
+      For model with non-fp32 input data type, add type suffix
+      ```shell
+      ./pnnx resnet18.pt inputshape=[1,3,224,224]f32,[1,32]i64
+      ```
+  3. pick resnet18_pnnx.py for pnnx-optimized torch model
+  4. pick resnet18.ncnn.param and resnet18.ncnn.bin for ncnn inference
+
+see more pnnx informations: https://github.com/pnnx/pnnx
 
+## pytorch to onnx
```
Comment (nihui) [inline docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md:111]:
```suggestion
## pytorch to onnx (deprecated)
```

Comment (whyb) [reply docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md]: this suggestion was successfully applied.

File: docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md (hunk 1 above)
Comment (whyb) [reply docs/how-to-use-and-FAQ/use-ncnn-with-pytorch-or-onnx.md]: this suggestion was successfully applied.

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**